packages/gatekeeper/src/helpers/services.ts

```diff
@@ -115,6 +119,13 @@ export const fetchState = async (
     })
     .then(data => {
       const { session } = data;
+      if (!session) {
+        logoutActionCreator();
+        const err = new Error('User is logged out');
+        recordResultActionCreator(false, { err });
+        authenticationProgressCreator(false);
+        return;
```

Comment:
Why are we calling return this way here?  Why not throw the error, or call `errorCallbackFn`?